editor/src/scene_viewer/gizmo.rs

```diff
@@ -297,9 +306,17 @@ impl SceneGizmo {
                 },
             )
         }
+        if left_mouse_button == ButtonState::Pressed {
+            let delta = pos - self.last_click_position;
+            if delta.magnitude() > 0.1 {
+                camera_controller.yaw = delta.x * 0.1;
```

Comment:
You also need to store initial values of yaw and pitch (at the click moment) and basically do this: `camera_controller.yaw = self.initial_yaw + delta.x * 0.1`. Otherwise you'll get jumping camera, which is already the case on the video you've attached.

Reply:
I am now storing the yaw and pitch when a left click technically on the scene is pressed but it also updates on clicking the gizmo. If I should place it further down I can but this seemed to make it better. 

I think another issue potentially is not having some element to grab onto so when trying to rotate the gizmo via click it first picks a node to switch perspective. 

I could place this functionality behind some kind of mesh sphere to help with this or if you have a suggestion there. Thanks for your help i've learned so much about the editor especially on this one 🙏

Reply:
All you need to do is to move interaction with arrows to mouse up event (instead of mouse down like it is now). It is located here - https://github.com/FyroxEngine/Fyrox/blob/master/editor/src/scene_viewer/mod.rs#L785, so you basically need to change `MouseDown` to `MouseUp` and add new match branch with `MouseDown` and put there rotation handling (its initiation, to be precise). Also, to prevent triggering of arrows on MouseUp when you end dragging, you may also add `is_dragging: bool` flag and raise it when you start dragging (when movement delta is large enough). Check this flag in `on_click` method and return if it is raised.